types/logger/logger.go

```diff
@@ -35,6 +36,23 @@ type Context context.Context
 
 type logfKey struct{}
 
+// J marshals v as JSON and writes it to logf formatted with the annotation to make logtail
+// treat it as a structured log.
+func J(logf Logf, recType string, v interface{}) {
+	// TODO: pool of json.Encoder+bytes.Buffer pre-wired pairs
```

Comment:
We have this for unmarshal in util/jsonutil, so it'd be a natural addition there. Although if I recall correctly, everyone hates that. :P And anyway, @dsnet is going to fix all our JSON performance issues. :)

Reply:
😏